### ISSUE TITLE
Why does kubent scan kubectl.kubernetes.io/last-applied-configuration annotation instead of the actual apiVersion

### ISSUE THREAD

**Issue by hpl002** [doitintl/kube-no-trouble#357]
This often leads to false positives.

**Comment by stepanstipl** [comment]
I just commented in the other issue, this is misunderstanding of how K8S API works. 

**TL;DR:** because there is no such thing as the "actual" API version. The API version that the resource is returned in is dictated by the client, can be any of the supported versions by the given K8S API server version (i.e. you can get same resource in 2 different versions), and does not reflect how the resource was created.

> Hi everyone,
> 
> so kubent does look **only** at the `last-applied-configuration` (or Helm in-cluster manifests if you use Helm). This is described in [README](https://github.com/doitintl/kube-no-trouble) and I've tried to explain it in https://github.com/doitintl/kube-no-trouble/issues/276#issuecomment-1062082004. This is due to how K8s work, and basically the original version used to create the resource is not preserved anywhere else (leaving aside newer managed fields for now).
> 
> As to `kubectl` - K8S will return the resource in the API version that was requested by the client, please see https://github.com/doitintl/kube-no-trouble/issues/276#issuecomment-1062928787. This version is independent of a) the version used when the resource was created and b) of the internal stored version.
> 
> **TL;DR:** If the `kubectl.kubernetes.io/last-applied-configuration` annotation contains old API version, it's expected behaviour this will be returned by kubent. This is how kubent works and I would say it's desirable, as it suggests the resource was created using the old API version, i.e. you will run into trouble if you try to do the same after cluster upgrade.
> 
> I can see that all the examples shared here contain the old API version in the annotation, so I'm going to close this issue. But if that's not the case, please comment. Thanks 👍
